src/index.mjs

```diff
@@ -62,7 +63,7 @@ export default function (options) {
 
   observer.priority = options.priority;
 
-  const allowed = options.origins || [];
+  const allowed = options.sameOrigin ? [location.hostname] : options.origins || [];
```

Comment:
We might want to make it clear (perhaps in the jsdoc) what the default behavior for `sameOrigin` is? :)

* Benefit of `sameOrigin` defaulting to true: avoids issues with CORB, fetch(), XHR cross-domain
* Benefit of it defaulting to false: still get the benefits of rel=prefetch'ing third-party links by default (this early on, it's hard to get a feel for how much sites benefit from this in practice).

Reply:
Sure! I can make a pass at the readme – I figured you'd drop some knowledge there anyway. 🎓 

This PR was just meant to match the current behavior. Technically, swapping `sameOrigin` to true would be a breaking change, I think. I'm generally hesitant towards major bumps so early on... but that's just me! 

As far as which way quicklink _should_ lean? No strong feelings, but if we are to keep both options, it's more config to disable sameOrigin restriction.


```js
// When sameorigin defaults true

// To turn off:
prefetch({
 origins: ['my', 'list', 'here'],
 sameOrigin: false
});
```

Another version (and maybe better?) is to work within a single `origin` option, which defaults to the sameOrigin initially. To turn _this_ off, you just re-define new `origin` list:

```js
// quicklink.js
opts = Object.assign({
  origin: [location.hostname]
}, opts);


// my-app.js
// Turn off sameOrigin
quicklink({ origin:false });
// or: { origin:[] }

// Defining custom origins also requires own origin be included:
quicklink({
  origin: [
    'my-app.com',
    'other-app.net'
  ]
})
```

Reply:
So, after https://github.com/GoogleChromeLabs/quicklink/issues/33#issuecomment-447174379, I think my vote is with this second option, along with a (breaking) change to restrict to same origin only.

This would mean `sameOrigin` is redundant & therefore removed. A combination of `origins` and `filter` should provide full control/customization – with `origins` exact-matching hostnames & `filter` handling the finer details. Similarly, one could opt-out of `origins` and move that work into their `filter` function(s) directly.

As suggested in that issue, when both `origins` and `filter` exist, the filter(s) should run _after_ the origin checks.

Reply:
After reviewing the discussion in #33, I'm strongly inclined to agree with the second option. Users appear to want control over `origins`, allow and disallow lists (which a  `filter` pretty much provides too). I'm happy for us to proceed with your suggested model here, @lukeed (thanks a lot for thinking through this too..).

A change here would fix both #33 and #31.